clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -627,6 +627,9 @@ private TransactionManager configureTransactionState(ProducerConfig config,
      * @throws KafkaException if the producer has encountered a previous fatal error or for any other unexpected error
      * @throws TimeoutException if the time taken for initialize the transaction has surpassed <code>max.block.ms</code>.
      * @throws InterruptException if the thread is interrupted while blocked
+     * @throws org.apache.kafka.common.errors.ClusterAuthorizationException fatal error indicating that the idempotent
```

Comment:
I think we can get rid of this since we have `AuthorizationException` above. Perhaps we can add the comment about retrying there?